src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -216,11 +216,11 @@ private void AfterConnectAcceptTelemetry()
             switch (LastOperation)
             {
                 case SocketAsyncOperation.Accept:
-                    SocketsTelemetry.Log.AfterAccept(SocketError);
+                    SocketsTelemetry.AfterAccept(SocketError);
```

Comment:
Yes, though all the core event logic is still guarded by `IsEnabled` checks.
What may now remain are the `Interlocked` calls mentioned above and this switch method.

If we wanted those to be trimmed out completely, I think we should expose the `EventSource.IsSupported` flag instead (currently internal), as that is what we actually care about here, not whether someone is listening at this exact moment.
Since `IsEnabled` can toggle between on/off, it can't be used the same way without complicating the calling logic.